docs/design/K2PROCESS.md

```diff
@@ -0,0 +1,29 @@
+This document describes the architecture for a K2 process.
+
+# General process architecture
+![architecture](../images/K2process.png)
+
+We use Seastar to manage our processes. When our process is started, we create N reactor instances - each instance pinned to a particular core. The number of instances and their core assignment is driven via command line configuration.
+
+The user code is packaged in Applets. An applet is simply any class which has a start and stop methods (see many examples under k2/src/cmd/demo)
```

Comment:
Maybe a note here that there is one instance of an Applet class per core/reactor.

Reply:
good point. expanded to include this